biojava-genome/src/test/java/org/biojava/nbio/genome/TestGenomeMapping.java

```diff
@@ -148,4 +145,34 @@ private void validateExon(int exonNr, int start, int stop, List<Range<Integer>>
         assertTrue("Exon " + exonNr + " boundary " + exon.upperEndpoint() + " does not match " + stop, exon.upperEndpoint().equals(stop));
 
     }
+    
+    @Test
+	/** Test to make sure the reverse mapping of a genomic coordinate handles border cases
+	 * 
+	 * @author Yana Valasatava
+	 */
+    public void testReverseMappingForExonBoundaries() {
+    	
+    	String geneName = "BCL11B"; // gene on the reverse DNA strand 
+    	String genebankId = "NM_138576"; // GeneBank ID for the transcript used for testing (ENST00000357195)
+    	
+    	int posExonStart = 99174151; // starting position of the first base in a coding region (1st exon)
+    	int posExonEnd = 99176195; // ending position of the first base in a coding region (1st exon)
+    	
+    	for (GeneChromosomePosition gcp : gcps) {
+    		
+    		if ( !gcp.getGeneName().equals(geneName) )
+    			continue;
+    		if ( !gcp.getGenebankId().equals(genebankId) )
+    			continue;
+    		
+    		int cdsSE = ChromosomeMappingTools.getCDSPosForChromosomeCoordinate(posExonStart-1, gcp);
+    		assertEquals(cdsSE, -1);
+    		
+    		int cdsEE = ChromosomeMappingTools.getCDSPosForChromosomeCoordinate(posExonEnd+1, gcp);
+    		assertEquals(cdsEE, -1);
+    		
```

Comment:
Can you also add a check that the CDS position for first base in the coding region is correct?